### PR TITLE
Update fields GiftCard Hub API

### PR DESCRIPTION
**Pull request by ricardoaerobr**
#### Types of changes
- [ ] New content (endpoints, descriptions or fields from scratch)
- [X] Improvement (make an endpoint's title or description even better)
- [ ] Spelling and grammar accuracy (self-explanatory)

### Changelog
Do not forget to update your changes to our Developer Portal's changelog. Did you create a release note?
- [ ] Yes, I already created a release note about this change.
- [X] No, but I am going to.

PR created to update descriptions of GiftCard Hub API endpoints (EDU-10571).

### CODE REVIEW

File: VTEX - Giftcard API.json [hunk 1]
```diff
@@ -1253,153 +1253,159 @@
...
-                            "refId": "12",
-                            "name": "Product Name",
-                            "price": 100,
-                            "quantity": 1
-                        }
-                    ],
-                    "itemsTotal": 100
-                }
-            },
+				"title": "Cart information.",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1256]: This component is used in a request body, therefore it must abide by the OAS guidelines for request bodies, please include examples to each field.

See our [openapi template](https://github.com/vtex/openapi-schemas/blob/master/templates/VTEX%20-%20Template%20openAPI.jsonc)

File: VTEX - Giftcard API.json [hunk 2]
```diff
@@ -1253,153 +1253,159 @@
...
+                        "description": "Represents the relationship between the client and the store."
+					},
+					"redemptionCode": {
+						"type": "string",
+						"default": "AH52F7",
+						"description": "Minimum of six characters"
+					},
+					"discounts": {
+						"type": "integer",
+						"format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1286]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 3]
```diff
@@ -1253,153 +1253,159 @@
...
+                        "description": "Discounts value."
+					},
+					"shipping": {
+						"type": "integer",
+						"format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1291]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 4]
```diff
@@ -1253,153 +1253,159 @@
...
+                        "description": "Shipping value."
+					},
+					"taxes": {
+						"type": "integer",
+						"format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1296]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 5]
```diff
@@ -1253,153 +1253,159 @@
...
+					"items": {
+						"type": "array",
+						"items": {
+							"$ref": "#/components/schemas/Item"
+						},
+						"description": ""
+					},
+					"itemsTotal": {
+						"type": "integer",
+						"format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1308]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 6]
```diff
@@ -1253,153 +1253,159 @@
...
-                "example": {
-                    "productId": "1",
-                    "id": "1",
-                    "refId": "12",
-                    "name": "Product Name",
-                    "price": 100,
-                    "quantity": 1
-                }
-            },
+				"title": "Items information.",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1333]: This component is used in a request body, therefore it must abide by the OAS guidelines for request bodies, please include examples to each field.

See our [openapi template](https://github.com/vtex/openapi-schemas/blob/master/templates/VTEX%20-%20Template%20openAPI.jsonc)

File: VTEX - Giftcard API.json [hunk 7]
```diff
@@ -1253,153 +1253,159 @@
...
+						"type": "string",
+                        "description": "The ID of the SKU in VTEX platform."
+					},
+					"refId": {
+						"type": "string",
+                        "description": "Product Reference ID."
+					},
+					"name": {
+						"type": "string",
+						"nullable": true,
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1358]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 8]
```diff
@@ -1253,153 +1253,159 @@
...
+                        "description": "Product name."
+					},
+					"price": {
+						"type": "integer",
+						"format": "int32",
+                        "description": "Product value."
+					},
+					"quantity": {
+						"type": "integer",
+						"format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1368]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 9]
```diff
@@ -1253,153 +1253,159 @@
...
+                        "description": "Product Reference ID."
+					},
+					"name": {
+						"type": "string",
+						"nullable": true,
+                        "description": "Product name."
+					},
+					"price": {
+						"type": "integer",
+						"format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1363]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 10]
```diff
@@ -1485,25 +1492,29 @@
...
             "Paging": {
-                "title": "Paging",
+                "title": "Pagination information.",
                 "required": ["page", "perPage", "total", "pages"],
                 "type": "object",
                 "properties": {
                     "page": {
                         "type": "integer",
-                        "format": "int32"
+                        "format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1501]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 11]
```diff
@@ -1485,25 +1492,29 @@
...
                     "perPage": {
                         "type": "integer",
-                        "format": "int32"
+                        "format": "int32",
+                        "description": "Quantity of gift cards per page."
                     },
                     "total": {
                         "type": "integer",
-                        "format": "int32"
+                        "format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1511]:
```suggestion
```

File: VTEX - Giftcard API.json [hunk 12]
```diff
@@ -1485,25 +1492,29 @@
...
+                        "description": "Total of gift cards in the store."
                     },
                     "pages": {
                         "type": "integer",
-                        "format": "int32"
+                        "format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1516]:
```suggestion
```

File: VTEX - GiftCard Hub API.json [hunk 13]
```diff
@@ -1238,16 +1238,43 @@
...
+                                            "type": "string",
+                                            "description": "Gift card identification."
+                                        },
+                                        "id": {
+                                            "type": "string",
+                                            "description": "Transaction identification."
+                                        },
+                                        "_self": {
+                                            "type": "object",
+                                            "description": "It is an object that carries an auto reference from the provider at the Hub (on its API).",
```
Comment (PedroAntunesCosta) [inline VTEX - GiftCard Hub API.json:1258]:
```suggestion
                                            "description": "Object that carries an auto reference from the provider at the Hub (on its API).",
```

File: VTEX - GiftCard Hub API.json [hunk 14]
```diff
@@ -2273,7 +2341,7 @@
                 }
             },
             "Item": {
```
Comment (PedroAntunesCosta) [inline VTEX - GiftCard Hub API.json:2343]: This component is used in a request body, therefore it must abide by the OAS guidelines for request bodies, please include examples to each field.

See our [openapi template](https://github.com/vtex/openapi-schemas/blob/master/templates/VTEX%20-%20Template%20openAPI.jsonc)

File: VTEX - GiftCard Hub API.json [hunk 15]
```diff
@@ -2327,30 +2395,35 @@
...
                     },
                     "orderInfo": {
                         "$ref": "#/components/schemas/OrderInfo"
                     }
                 },
-                "default": {
+                "example": {
                     "operation": "Credit",
                     "value": 28,
                     "description": "example",
```
Comment (PedroAntunesCosta) [inline VTEX - GiftCard Hub API.json:2429]: It would be nice to have an actual example here.

File: VTEX - GiftCard Hub API.json [hunk 16]
```diff
@@ -2407,22 +2480,17 @@
...
                 "properties": {
                     "orderId": {
-                        "type": "string"
+                        "type": "string",
+                        "description": "Order ID."
                     },
                     "sequence": {
                         "type": "integer",
-                        "format": "int32"
+                        "format": "int32",
```
Comment (PedroAntunesCosta) [inline VTEX - GiftCard Hub API.json:2492]:
```suggestion
```

File: VTEX - GiftCard Hub API.json [hunk 17]
```diff
@@ -2434,56 +2502,60 @@
                         "$ref": "#/components/schemas/Shipping"
                     }
                 },
-                "example": {
-                    "orderId": "v2915869org-01",
-                    "sequence": 2915869,
+                "default": {
```
Comment (PedroAntunesCosta) [inline VTEX - GiftCard Hub API.json:2505]:
```suggestion
                "example": {
```

Comment (PedroAntunesCosta) [reply VTEX - GiftCard Hub API.json:2505]: Only use `default` for fields that:
- Are not required.
- Are in request the request body.
- Have a default value that influences the platform behavior if you don't send a custom one in the request.

File: VTEX - GiftCard Hub API.json [hunk 18]
```diff
@@ -2434,56 +2502,60 @@
...
-                        "reference": ""
+                        "street": "Praia de Botafogo",
+                        "number": "11",
+                        "neighborhood": "example",
+                        "complement": null,
+                        "reference": null
                     }
                 }
             },
             "Cart1": {
```
Comment (PedroAntunesCosta) [inline VTEX - GiftCard Hub API.json:2557]: This component is used in a request body, therefore it must abide by the OAS guidelines for request bodies, please include examples to each field.

See our [openapi template](https://github.com/vtex/openapi-schemas/blob/master/templates/VTEX%20-%20Template%20openAPI.jsonc)

**Review by PedroAntunesCosta — CHANGES_REQUESTED** [review]

File: VTEX - Giftcard API.json [hunk 19]
```diff
@@ -1214,7 +1214,7 @@
                 }
             },
             "GetGiftCardusingJSONRequest": {
-                "title": "GetGiftCardusingJSONRequest",
+                "title": "",
```
[the remark this answers is not in the document]
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1217]: Agree.

File: VTEX - Giftcard API.json (hunk 2 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1285]: Corrigido.

File: VTEX - Giftcard API.json (hunk 4 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1295]: Corrigido.

File: VTEX - Giftcard API.json (hunk 8 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1367]: Corrigido.

File: VTEX - Giftcard API.json (hunk 9 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1362]: Corrigido.

File: VTEX - Giftcard API.json (hunk 10 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1500]: Corrigido.

File: VTEX - Giftcard API.json (hunk 11 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1510]: Corrigido.

File: VTEX - Giftcard API.json (hunk 12 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1515]: Corrigido.

File: VTEX - GiftCard Hub API.json (hunk 13 above)
Comment (ricardoaerobr) [reply VTEX - GiftCard Hub API.json:1258]: Corrigido.

File: VTEX - Giftcard API.json (hunk 1 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1255]: Corrigido.

File: VTEX - Giftcard API.json (hunk 6 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1332]: Corrigido.

File: VTEX - Giftcard API.json (hunk 7 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json:1357]: Corrigido.

File: VTEX - GiftCard Hub API.json (hunk 18 above)
Comment (ricardoaerobr) [reply VTEX - GiftCard Hub API.json:2557]: Corrigido.

File: VTEX - Giftcard API.json [hunk 20]
```diff
@@ -1253,153 +1251,163 @@
...
+						"type": "string",
+						"nullable": true,
+                        "description": "Represents the relationship between the client and the store.",
+                        "example": null
+					},
+					"redemptionCode": {
+						"type": "string",
+						"default": "AH52F7",
+						"description": "Minimum of six characters",
+                        "example": "XTYB"
```
Comment (PedroAntunesCosta) [inline VTEX - Giftcard API.json:1282]: - Doesn't it have a minimum of six characters?

**Review by PedroAntunesCosta — CHANGES_REQUESTED** [review]

File: VTEX - Giftcard API.json (hunk 20 above)
Comment (ricardoaerobr) [reply VTEX - Giftcard API.json]: Adjusted.

**Review by PedroAntunesCosta — APPROVED** [review]
LGTM

**Pull request merged**